text/label/subword_nmt/train.py

```diff
@@ -95,33 +95,51 @@ class ReturnnTrainBpeJob(Job):
     """
     Create Bpe codes and vocab files compatible with RETURNN BytePairEncoding
     Repository:
-        https://github.com/albertz/subword-nmt
+        https://github.com/rwth-i6/subword-nmt
 
     This job can be used to produce BPE codes compatible to legacy (non-sisyphus) RETURNN setups.
+
+    Outputs:
+        - bpe_codes: the codes file to apply BPE to any text
+        - bpe_vocab: the index vocab in the form of {"<token>": <index>, ...} that can be used e.g. for RETURNN
+            Will contain <s> and </s> pointing to index 0 and the unk_label pointing to index 1
+        - bpe_apply_vocab: a text file containing all words, to be used with the `ApplyBPEToTextJob`
+            DOES NOT INCLUDE COUNTS, but just set each count to 100. Is used to not cause invalid merges
+            when converting text to the BPE form.
+        - vocab_size: variable containing the number of indices
     """
 
-    def __init__(self, text_file, bpe_size, unk_label="UNK", subword_nmt_repo=None):
+    def __init__(
+        self,
+        text_file: tk.Path,
+        bpe_size: int,
+        unk_label: str = "UNK",
+        subword_nmt_repo: Optional[tk.Path] = None,
+    ):
         """
-        :param Path text_file: corpus text file
+        :param text_file: corpus text file, only accepts uncompressed text
```

Comment:
why does it only support uncompressed text files? Below `util.uopen` is used to open the file which should handle gzipped files correctly.

Reply:
Yes this is a leftover because I first added this comment, and then added the automatic uncompressing beforehand. So this can be removed now.